ergo/distributions/conditions.py

```diff
@@ -57,3 +60,31 @@ def loss(self, dist):
 
     def __str__(self):
         return f"The probability density function looks similar to the provided density function."
+
+
+@dataclass
+class IntervalCondition(Condition):
+    p: float
+    low: Optional[float]
+    high: Optional[float]
+    weight: float
+
+    def __init__(self, p, low=None, high=None, weight=1.0):
```

Comment:
the value for one end being open should maybe be +-infinity, rather than None

wasn't sure how likely infinity was to be annoying compared to None, I think both sort of make sense

Reply:
(switched to infinity, seems better I think)